types/codemirror/addon/hint/show-hint.d.ts

```diff
@@ -54,8 +54,13 @@ declare module "codemirror" {
     }
 
     interface ShowHintOptions {
-        completeSingle: boolean;
+        completeSingle?: boolean;
         hint: HintFunction | AsyncHintFunction;
+        alignWithWord?: boolean;
+        closeOnUnfocus?: boolean;
+        container?: Element | null;
+        customKeys?: {[x: string]: Function};
+        extraKeys?: {[x: string]: Function};
```

Comment:
`customKeys` and `extraKeys` would be nullable according to https://github.com/codemirror/CodeMirror/blob/5d740b99a4cca996b2cdc9fee3b1e1b8ac999617/addon/hint/show-hint.js#L461-L462